js/util/verticalize_punctuation.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const scriptDetection = require('./script_detection');
+
+module.exports = function verticalizePunctuation(input) {
+    let output = '';
+
+    for (let i = 0; i < input.length; i++) {
+        const nextCharCode = input.charCodeAt(i + 1) || null;
+        const prevCharCode = input.charCodeAt(i - 1) || null;
+
+        const canReplacePunctuation = (
+            (!nextCharCode || scriptDetection.charAllowsVerticalWritingMode(nextCharCode, true)) &&
+            (!prevCharCode || scriptDetection.charAllowsVerticalWritingMode(prevCharCode, true))
+        );
+
+        if (canReplacePunctuation && module.exports.lookup[input[i]]) {
+            output += module.exports.lookup[input[i]];
+        } else {
+            output += input[i];
+        }
+    }
+
+    return output;
+};
+
+module.exports.lookup = {
+    '。': '︒',
+    '—': '︱',
+    '–': '︲',
+    '_': '︳',
+    '（': '︵',
+    '）': '︶',
+    '｛': '︷',
+    '｝': '︸',
+    '〔': '︹',
+    '〕': '︺',
+    '〘': '︹',
+    '〙': '︺',
+    '【': '︻',
+    '】': '︼',
+    '《': '︽',
+    '》': '︾',
+    '〈': '︿',
+    '〉': '﹀',
+    '「': '﹁',
+    '」': '﹂',
+    '『': '﹃',
+    '』': '﹄',
+    '｢': '﹁',
+    '｣': '﹂',
+    '［': '﹇',
+    '］': '﹈',
+    '“': '﹁',
+    '”': '﹂',
+    '‘': '﹃',
+    '’': '﹄',
+    '.': '・'
```

Comment:
This comment was lost in the rebase cc @1ec5 @nickidlugash 

<img width="611" alt="screen shot 2016-11-09 at 4 50 04 pm" src="https://cloud.githubusercontent.com/assets/281306/20160556/9ec72800-a69c-11e6-894f-565f527e197a.png">